### ISSUE TITLE
Feature request: provide more granularity to disable notifications that cannot be fixed

### ISSUE THREAD

**Issue by eternoendless** [composer/packagist#1193]
We recently submitted [PrestaShop](https://github.com/PrestaShop/PrestaShop/) to Packagist to simplify the way some people create their new projects using composer, and it works great.

Unfortunately, we are receiving the following notifications several times a day:

> The prestashop/prestashop package of which you are a maintainer has
> failed to update due to invalid data contained in your composer.json.
> Please address this as soon as possible since the package stopped updating.
> 
> It is recommended that you use `composer validate` to check for errors when you
> change your composer.json.
> 
> Below is the full update log which should highlight errors as
> "Skipped branch ...":
> 
> [Composer\Repository\InvalidRepositoryException]: Some branches contained invalid data and were discarded, it is advised to review the log and fix any issues present in branches
> 
> Reading composer.json of prestashop/prestashop (1.7.7.4)
> Found cached composer.json of prestashop/prestashop (1.7.7.4)
> Reading composer.json of prestashop/prestashop (1.7.7.3)
> Found cached composer.json of prestashop/prestashop (1.7.7.3)
> Reading composer.json of prestashop/prestashop (1.7.7.2)
> Found cached composer.json of prestashop/prestashop (1.7.7.2)
> Reading composer.json of prestashop/prestashop (1.7.7.1)
> Found cached composer.json of prestashop/prestashop (1.7.7.1)
> Reading composer.json of prestashop/prestashop (1.7.7.0)
> Found cached composer.json of prestashop/prestashop (1.7.7.0)
> Reading composer.json of prestashop/prestashop (1.7.7.0-rc.1)
> Found cached composer.json of prestashop/prestashop (1.7.7.0-rc.1)
> Reading composer.json of prestashop/prestashop (1.7.7.0-beta.2)
> Found cached composer.json of prestashop/prestashop (1.7.7.0-beta.2)
> Reading composer.json of prestashop/prestashop (1.7.7.0-beta.1)
> Found cached composer.json of prestashop/prestashop (1.7.7.0-beta.1)
> Reading composer.json of prestashop/prestashop (1.7.6.9)
> Found cached composer.json of prestashop/prestashop (1.7.6.9)
> Reading composer.json of prestashop/prestashop (1.7.6.8)
> Found cached composer.json of prestashop/prestashop (1.7.6.8)
> Reading composer.json of prestashop/prestashop (1.7.6.7)
> Found cached composer.json of prestashop/prestashop (1.7.6.7)
> Reading composer.json of prestashop/prestashop (1.7.6.6)
> Found cached composer.json of prestashop/prestashop (1.7.6.6)
> Reading composer.json of prestashop/prestashop (1.7.6.5)
> Found cached composer.json of prestashop/prestashop (1.7.6.5)
> Reading composer.json of prestashop/prestashop (1.7.6.4)
> Found cached composer.json of prestashop/prestashop (1.7.6.4)
> Reading composer.json of prestashop/prestashop (1.7.6.3)
> Found cached composer.json of prestashop/prestashop (1.7.6.3)
> Reading composer.json of prestashop/prestashop (1.7.6.2)
> Found cached composer.json of prestashop/prestashop (1.7.6.2)
> Reading composer.json of prestashop/prestashop (1.7.6.1)
> Found cached composer.json of prestashop/prestashop (1.7.6.1)
> Reading composer.json of prestashop/prestashop (1.7.6.0)
> Found cached composer.json of prestashop/prestashop (1.7.6.0)
> Reading composer.json of prestashop/prestashop (1.7.6.0-rc.2)
> Found cached composer.json of prestashop/prestashop (1.7.6.0-rc.2)
> Reading composer.json of prestashop/prestashop (1.7.6.0-rc.1)
> Found cached composer.json of prestashop/prestashop (1.7.6.0-rc.1)
> Reading composer.json of prestashop/prestashop (1.7.6.0-beta.1)
> Found cached composer.json of prestashop/prestashop (1.7.6.0-beta.1)
> Reading composer.json of prestashop/prestashop (1.7.5.2)
> Found cached composer.json of prestashop/prestashop (1.7.5.2)
> Reading composer.json of prestashop/prestashop (1.7.5.1)
> Found cached composer.json of prestashop/prestashop (1.7.5.1)
> Reading composer.json of prestashop/prestashop (1.7.5.0)
> Found cached composer.json of prestashop/prestashop (1.7.5.0)
> Reading composer.json of prestashop/prestashop (1.7.5.0-rc.1)
> Found cached composer.json of prestashop/prestashop (1.7.5.0-rc.1)
> Reading composer.json of prestashop/prestashop (1.7.5.0-beta.1)
> Found cached composer.json of prestashop/prestashop (1.7.5.0-beta.1)
> Reading composer.json of prestashop/prestashop (1.7.4.4)
> Found cached composer.json of prestashop/prestashop (1.7.4.4)
> Reading composer.json of prestashop/prestashop (1.7.4.3)
> Found cached composer.json of prestashop/prestashop (1.7.4.3)
> Reading composer.json of prestashop/prestashop (1.7.4.2)
> Found cached composer.json of prestashop/prestashop (1.7.4.2)
> Reading composer.json of prestashop/prestashop (1.7.4.1)
> Found cached composer.json of prestashop/prestashop (1.7.4.1)
> Reading composer.json of prestashop/prestashop (1.7.4.0)
> Found cached composer.json of prestashop/prestashop (1.7.4.0)
> Reading composer.json of prestashop/prestashop (1.7.4.0-beta.1)
> Found cached composer.json of prestashop/prestashop (1.7.4.0-beta.1)
> Reading composer.json of prestashop/prestashop (1.7.3.4)
> Found cached composer.json of prestashop/prestashop (1.7.3.4)
> Reading composer.json of prestashop/prestashop (1.7.3.3)
> Found cached composer.json of prestashop/prestashop (1.7.3.3)
> Reading composer.json of prestashop/prestashop (1.7.3.2)
> Found cached composer.json of prestashop/prestashop (1.7.3.2)
> Reading composer.json of prestashop/prestashop (1.7.3.1)
> Found cached composer.json of prestashop/prestashop (1.7.3.1)
> Reading composer.json of prestashop/prestashop (1.7.3.0)
> Found cached composer.json of prestashop/prestashop (1.7.3.0)
> Reading composer.json of prestashop/prestashop (1.7.2.5)
> Found cached composer.json of prestashop/prestashop (1.7.2.5)
> Reading composer.json of prestashop/prestashop (1.7.2.4)
> Found cached composer.json of prestashop/prestashop (1.7.2.4)
> Reading composer.json of prestashop/prestashop (1.7.2.3)
> Found cached composer.json of prestashop/prestashop (1.7.2.3)
> Reading composer.json of prestashop/prestashop (1.7.2.2)
> Found cached composer.json of prestashop/prestashop (1.7.2.2)
> Reading composer.json of prestashop/prestashop (1.7.2.1)
> Found cached composer.json of prestashop/prestashop (1.7.2.1)
> Reading composer.json of prestashop/prestashop (1.7.2.0)
> Found cached composer.json of prestashop/prestashop (1.7.2.0)
> Reading composer.json of prestashop/prestashop (1.7.2.0-rc.1.0)
> Found cached composer.json of prestashop/prestashop (1.7.2.0-rc.1.0)
> Reading composer.json of prestashop/prestashop (1.7.1.2)
> Found cached composer.json of prestashop/prestashop (1.7.1.2)
> Reading composer.json of prestashop/prestashop (1.7.1.1)
> Found cached composer.json of prestashop/prestashop (1.7.1.1)
> Reading composer.json of prestashop/prestashop (1.7.1.0)
> Found cached composer.json of prestashop/prestashop (1.7.1.0)
> Reading composer.json of prestashop/prestashop (1.7.0.6)
> Found cached composer.json of prestashop/prestashop (1.7.0.6)
> Reading composer.json of prestashop/prestashop (1.7.0.5)
> Found cached composer.json of prestashop/prestashop (1.7.0.5)
> Reading composer.json of prestashop/prestashop (1.7.0.4)
> Found cached composer.json of prestashop/prestashop (1.7.0.4)
> Reading composer.json of prestashop/prestashop (1.7.0.3)
> Found cached composer.json of prestashop/prestashop (1.7.0.3)
> Reading composer.json of prestashop/prestashop (1.7.0.2)
> Found cached composer.json of prestashop/prestashop (1.7.0.2)
> Reading composer.json of prestashop/prestashop (1.7.0.1)
> Found cached composer.json of prestashop/prestashop (1.7.0.1)
> Reading composer.json of prestashop/prestashop (1.7.0.0)
> Found cached composer.json of prestashop/prestashop (1.7.0.0)
> Reading composer.json of prestashop/prestashop (1.7.0.0-rc.2.0)
> Found cached composer.json of prestashop/prestashop (1.7.0.0-rc.2.0)
> Reading composer.json of prestashop/prestashop (1.7.0.0-rc.1.0)
> Found cached composer.json of prestashop/prestashop (1.7.0.0-rc.1.0)
> Reading composer.json of prestashop/prestashop (1.7.0.0-rc.0.0)
> Found cached composer.json of prestashop/prestashop (1.7.0.0-rc.0.0)
> Reading composer.json of prestashop/prestashop (1.7.0.0-beta.4.0)
> Found cached composer.json of prestashop/prestashop (1.7.0.0-beta.4.0)
> Reading composer.json of prestashop/prestashop (1.7.0.0-beta.3.0)
> Found cached composer.json of prestashop/prestashop (1.7.0.0-beta.3.0)
> Reading composer.json of prestashop/prestashop (1.7.0.0-beta.2.0)
> Found cached composer.json of prestashop/prestashop (1.7.0.0-beta.2.0)
> Reading composer.json of prestashop/prestashop (1.7.0.0-beta.1.0)
> Found cached composer.json of prestashop/prestashop (1.7.0.0-beta.1.0)
> Reading composer.json of prestashop/prestashop (1.6.1.24)
> Skipped tag 1.6.1.24, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.23)
> Skipped tag 1.6.1.23, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.22)
> Skipped tag 1.6.1.22, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.21)
> Skipped tag 1.6.1.21, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.20)
> Skipped tag 1.6.1.20, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.19)
> Skipped tag 1.6.1.19, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.18)
> Skipped tag 1.6.1.18, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.17)
> Skipped tag 1.6.1.17, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.16)
> Skipped tag 1.6.1.16, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.15)
> Skipped tag 1.6.1.15, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.14)
> Skipped tag 1.6.1.14, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.13)
> Skipped tag 1.6.1.13, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.12)
> Skipped tag 1.6.1.12, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.11)
> Skipped tag 1.6.1.11, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.11-beta.1.0)
> Skipped tag 1.6.1.11-beta.1.0, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.10)
> Skipped tag 1.6.1.10, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.9)
> Skipped tag 1.6.1.9, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.8)
> Skipped tag 1.6.1.8, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.7)
> Skipped tag 1.6.1.7, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.6)
> Skipped tag 1.6.1.6, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.5)
> Skipped tag 1.6.1.5, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.4)
> Skipped tag 1.6.1.4, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.3)
> Skipped tag 1.6.1.3, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.2)
> Skipped tag 1.6.1.2, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.2-RC4)
> Skipped tag 1.6.1.2-RC4, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.2-RC3)
> Skipped tag 1.6.1.2-RC3, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.1)
> Skipped tag 1.6.1.1, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.1.0)
> Skipped tag 1.6.1.0, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.0.14)
> Skipped tag 1.6.0.14, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.0.13)
> Skipped tag 1.6.0.13, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.0.12)
> Skipped tag 1.6.0.12, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.0.11)
> Skipped tag 1.6.0.11, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.0.10)
> Skipped tag 1.6.0.10, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.0.9)
> Skipped tag 1.6.0.9, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.0.8)
> Skipped tag 1.6.0.8, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.0.7)
> Skipped tag 1.6.0.7, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.0.6)
> Skipped tag 1.6.0.6, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.0.5)
> Skipped tag 1.6.0.5, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.0.4)
> Skipped tag 1.6.0.4, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.0.3)
> Skipped tag 1.6.0.3, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.0.2)
> Skipped tag 1.6.0.2, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.6.0.1)
> Skipped tag 1.6.0.1, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.6.2)
> Skipped tag 1.5.6.2, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.6.1)
> Skipped tag 1.5.6.1, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.6.0)
> Skipped tag 1.5.6.0, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.5.0)
> Skipped tag 1.5.5.0, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.4.1)
> Skipped tag 1.5.4.1, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.4.0)
> Skipped tag 1.5.4.0, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.3.1)
> Skipped tag 1.5.3.1, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.3.0)
> Skipped tag 1.5.3.0, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.2.0)
> Skipped tag 1.5.2.0, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.1.0)
> Skipped tag 1.5.1.0, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.0.17)
> Skipped tag 1.5.0.17, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.0.15)
> Skipped tag 1.5.0.15, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.0.13)
> Skipped tag 1.5.0.13, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.0.9)
> Skipped tag 1.5.0.9, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.0.5)
> Skipped tag 1.5.0.5, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.0.3)
> Skipped tag 1.5.0.3, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.0.2)
> Skipped tag 1.5.0.2, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.0.1)
> Skipped tag 1.5.0.1, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (1.5.0.0)
> Skipped tag 1.5.0.0, no composer file was found (404 HTTP status code)
> Reading composer.json of prestashop/prestashop (develop)
> Importing branch develop (dev-develop)
> Reading composer.json of prestashop/prestashop (1.7.0.x)
> Importing branch 1.7.0.x (1.7.0.x-dev)
> Skipped branch 1.7.0.x, Invalid package information:
> Deprecation warning: require.beberlei/DoctrineExtensions is invalid, it should not contain uppercase characters. Please use beberlei/doctrineextensions instead. Make sure you fix this as Composer 2.0 will error.
> 
> Reading composer.json of prestashop/prestashop (1.7.1.x)
> Importing branch 1.7.1.x (1.7.1.x-dev)
> Skipped branch 1.7.1.x, Invalid package information:
> Deprecation warning: require.beberlei/DoctrineExtensions is invalid, it should not contain uppercase characters. Please use beberlei/doctrineextensions instead. Make sure you fix this as Composer 2.0 will error.
> 
> Reading composer.json of prestashop/prestashop (1.7.2.x)
> Importing branch 1.7.2.x (1.7.2.x-dev)
> Skipped branch 1.7.2.x, Invalid package information:
> Deprecation warning: require.beberlei/DoctrineExtensions is invalid, it should not contain uppercase characters. Please use beberlei/doctrineextensions instead. Make sure you fix this as Composer 2.0 will error.
> 
> Reading composer.json of prestashop/prestashop (1.7.3.x)
> Importing branch 1.7.3.x (1.7.3.x-dev)
> Skipped branch 1.7.3.x, Invalid package information:
> Deprecation warning: require.beberlei/DoctrineExtensions is invalid, it should not contain uppercase characters. Please use beberlei/doctrineextensions instead. Make sure you fix this as Composer 2.0 will error.
> 
> Reading composer.json of prestashop/prestashop (1.7.4.x)
> Importing branch 1.7.4.x (1.7.4.x-dev)
> Skipped branch 1.7.4.x, Invalid package information:
> Deprecation warning: require.beberlei/DoctrineExtensions is invalid, it should not contain uppercase characters. Please use beberlei/doctrineextensions instead. Make sure you fix this as Composer 2.0 will error.
> 
> Reading composer.json of prestashop/prestashop (1.7.5.x)
> Importing branch 1.7.5.x (1.7.5.x-dev)
> Skipped branch 1.7.5.x, Invalid package information:
> Deprecation warning: require.beberlei/DoctrineExtensions is invalid, it should not contain uppercase characters. Please use beberlei/doctrineextensions instead. Make sure you fix this as Composer 2.0 will error.
> 
> Reading composer.json of prestashop/prestashop (1.7.6.x)
> Found cached composer.json of prestashop/prestashop (1.7.6.x-dev)
> Reading composer.json of prestashop/prestashop (1.7.7.x)
> Found cached composer.json of prestashop/prestashop (1.7.7.x-dev)
> Reading composer.json of prestashop/prestashop (1.7.8.x)
> Importing branch 1.7.8.x (1.7.8.x-dev)

It is normal that some branches or tags are invalid and need to be skipped, they are historical and abandoned. However, there is no way to turn off _only_ those notifications: it's all or nothing.

Could it be possible to receive one and only one notification for each fail? Packagist could track the git ref of the branch/tag that failed, so that we don't get notified again if the ref is unchanged. Example:

First run: someproject/someproject tag 1.2.3 fails
	- see last checked ref for this tag (null)
	- send notification 
	- save ref for this tag (3a7fb13f50)
Second run: someproject/someproject tag 1.2.3 fails
	- see last checked ref for this tag (3a7fb13f50)
	- see current ref (3a7fb13f50)
	- the ref is still the same, don't send notification again

Alternatively, it would be great to be able to uncheck tags/branches individually.

**Comment by Seldaek** [comment]
The notifications are only sent if a branch fails to update, because these are actually fixable problems. Ignored tags are just output for information and are not considered a problem as those cannot be updated.

So I suggest you fix these warnings on the following branches:

```
Importing branch 1.7.0.x (1.7.0.x-dev)
Deprecation warning: require.beberlei/DoctrineExtensions is invalid, it should not contain uppercase characters. Please use beberlei/doctrineextensions instead. Make sure you fix this as Composer 2.0 will error.

Importing branch 1.7.1.x (1.7.1.x-dev)
Deprecation warning: require.beberlei/DoctrineExtensions is invalid, it should not contain uppercase characters. Please use beberlei/doctrineextensions instead. Make sure you fix this as Composer 2.0 will error.

Importing branch 1.7.2.x (1.7.2.x-dev)
Deprecation warning: require.beberlei/DoctrineExtensions is invalid, it should not contain uppercase characters. Please use beberlei/doctrineextensions instead. Make sure you fix this as Composer 2.0 will error.

Importing branch 1.7.3.x (1.7.3.x-dev)
Deprecation warning: require.beberlei/DoctrineExtensions is invalid, it should not contain uppercase characters. Please use beberlei/doctrineextensions instead. Make sure you fix this as Composer 2.0 will error.

Importing branch 1.7.4.x (1.7.4.x-dev)
Deprecation warning: require.beberlei/DoctrineExtensions is invalid, it should not contain uppercase characters. Please use beberlei/doctrineextensions instead. Make sure you fix this as Composer 2.0 will error.

Importing branch 1.7.5.x (1.7.5.x-dev)
Deprecation warning: require.beberlei/DoctrineExtensions is invalid, it should not contain uppercase characters. Please use beberlei/doctrineextensions instead. Make sure you fix this as Composer 2.0 will error.
```

Once that is done you shouldn't see any more warnings. And it's just a case change so completely harmless to do.

**Comment by eternoendless** [comment]
Fixed by deleting the offending branches 👍
